Sources/Gatekeeper/Gatekeeper.swift

```diff
@@ -11,29 +11,35 @@ public struct Gatekeeper {
         self.keyMaker = identifier
     }
     
-    public func gatekeep(on req: Request) -> EventLoopFuture<Void> {
+    public func gatekeep(
+        on req: Request,
+        throwing error: Error = Abort(.tooManyRequests, reason: "Slow down. You sent too many requests.")
+    ) -> EventLoopFuture<Void> {
         keyMaker
             .make(for: req)
             .flatMap { cacheKey in
                 fetchOrCreateEntry(for: cacheKey, on: req)
+                    .guard(
+                        { $0.requestsLeft > 0 },
+                        else: error
+                    )
                     .map(updateEntry)
-                    .flatMap { entry in
-                        cache
-                            .set(cacheKey, to: entry)
+                    .flatMap { entry -> EventLoopFuture<Entry> in
```

Comment:
This can just return future of void now and the transforms below can be removed.

Reply:
Ah yes, well spotted!